### ISSUE TITLE
[QA] Define project conventions

### ISSUE THREAD

**Issue by luzelenatt** [unmsmfisi-socialapplication/social_app#5]
Code: english git workflow: select workflow according to https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow 
commit format: 
reviews: at least one from some teammate Main branches (main, dev, etc) are read-only, no direct write

**Comment by luzelenatt** [comment]
A group meeting was held where various aspects of the first activity of the QA Team were discussed.

The following Google Docs document is attached: https://docs.google.com/document/d/1FOZTON4e3cc5SUxwjUaN8cFJXfgbVXNJPfgwSyGvLZA/edit?usp=sharing 

**WORKFLOW:**

**1. Creation of Main Branches:**
- A **main** branch is created, representing the stable version, and a **develop** branch is derived from it, which serves as the continuous integration branch.

**2. Feature Development (Feature Branches):**
- Each team (Backend, Frontend, Data, UX/UI, Mobile) creates feature branches derived from develop.

     BE: BACKEND
     FE: FRONTEND
     DATA: DATA
     UX-UI: UX/UI
     MOBILE: MOBILE
     QA: QA
     DEVOPS: DEVOPS

<html>
<body>
<b style="font-weight:normal;" id="docs-internal-guid-e78640e3-7fff-cf39-c31e-c030f0259362"><div dir="ltr" style="margin-left:-25.5pt;" align="left">

<html>
<body>


NOMENCLATURE | EXAMPLE | EXPLANATION
-- | -- | --
[ROLE]_[ACTIVITY NAME] | BE_FEATURE1 | To develop a new feature and/or activity. Work is done in this branch, and regular commits are made.


</body>
</html>

</div></b>
</body>
</html>

- QA can create their own branch derived from the feature branch for early testing.

**3. Feature Integration:**
- Once completed, feature branches are merged into the **develop** branch. 
- QA performs integration testing on the QA feature branch.

**4. Release Branch Creation:**
- When planning a new release, a release branch is created derived from **develop**.

**5. Release Testing:**
- QA conducts comprehensive testing on the release branch to ensure stability and quality before release.

**6. Integration into Main and Develop:**
- When release branch testing is successful, it is merged into both **main** and **develop**.

**7. Hotfix Management:**
- If a critical issue is found in the **main** branch, a hotfix branch is created derived from **main**.

**8. Hotfix Resolution and Testing:**
- The relevant team (Backend, Frontend, etc.) will address the issue in the **hotfix** branch, and QA will conduct testing on this branch.

**9. Integration into Main and Develop:**
- Once the hotfix is tested and approved, it is merged into both **main** and **develop**.

**COMMIT FORMAT:**
- feat: New feature
- fix: Bug fix
- docs: Documentation changes
- style: Formatting changes (no functional changes)
- refactor: Code changes that do not add features or fix bugs
- test: Adding or improving tests
- chore: Maintenance tasks and other non-code-related tasks

Example:
[Role] <type>: <brief description>
Optional body to describe project progress.
References: #IssueNumber, @Username

[Backend] fix: Fix the database
Resolved an error that was preventing user registrations on the social media platform.
References: #123, @backend_user

**REVISIONS:**
A table was created to determine which team member will be responsible for reviewing progress for each established team.

<html>
<body>
<b style="font-weight:normal;" id="docs-internal-guid-5ec0a561-7fff-56cd-6849-aa2ae9287d52"><div dir="ltr" style="margin-left:0pt;" align="left">

Team | Review Owner | Review Frequency
-- | -- | --
Data | Angel Falcon | Every two weeks
Back-end | Andrea Huarhuachi y Paula Rojas
Front-end | Cristhel Quito
UX/UI | Luz Torres
Mobile | Felix Fernandez
Devops | Gustavo Alania

</div><br /></b>
</body>
</html>


**DEFINED BRANCHES:**
**1. Main Branch:**
main: This branch contains stable and production-ready versions. No direct changes are made to this branch except for final versions. Merges into this branch occur when a version is complete and tested.

**2. Development Branch:**
develop: This is the main development branch. New features and changes are integrated into this branch. Merges into this branch are frequent and contain the latest version in development.

**3. Feature Branches:**
feature/feature-name: Each new feature is developed in its own feature branch derived from develop. Once complete, it is merged back into develop.

**4. Release Branches:**
release/version: Before a scheduled release, a release branch is created from develop. Final adjustments, testing, and minor fixes can be made here. Once ready for release, it is merged into both main and develop and tagged with the version number.

**5. Hotfix Branches:**
hotfix/version: If a critical error is discovered in the production version, a hotfix branch is created from main. The error is fixed and merged back into both main and develop, and tagged with the correction version.

**Comment by eds000n** [comment]
See this regaring git branching strategies https://www.linkedin.com/pulse/demystifying-git-branching-strategies-finding-right-your-meikopoulos/ @unmsmfisi-socialapplication/qa

**Comment by eds000n** [comment]
I opened a PR and I can merge with no review in `develop` branch; please change the configurations of the repo to set what you've decided. @felixalfonsofernandez

**Comment by luzelenatt** [comment]
The proofreader will be assigned to @Cristhel17

**Comment by Cristhel17** [comment]
Already we forked it to try a pull request in branch develop and teacher ,you're right, but we don't have the project configuration option to make the change because we haven't created the repository, attached image. @eds000n.

![image](https://github.com/unmsmfisi-socialapplication/social_app/assets/80082868/a5d7c6b8-0535-4d23-b49b-3b7096c4dcd5)

**Comment by felixalfonsofernandez** [comment]
![Image](https://github.com/unmsmfisi-socialapplication/social_app/assets/80082787/d32274d7-baf5-4202-b4da-c9a18ee12a61)

already i've permissions, we have granted the options to develop,

**Comment by felixalfonsofernandez** [comment]
@luzelenatt 
already made a restrict for main branch
